LazyProxy/LazyProxyBuilder.cs

```diff
@@ -206,5 +213,43 @@ private static MethodInfo GetGetServiceValueMethod(FieldInfo serviceField)
             // ReSharper disable once PossibleNullReferenceException
             return serviceField.FieldType.GetProperty("Value").GetGetMethod(true);
         }
+
+        private static void AddGenericParameters(
+            Func<IReadOnlyList<Type>> getGenericParameters,
+            Func<string[], IReadOnlyList<GenericTypeParameterBuilder>> defineGenericParameters)
+        {
+            var genericParameters = getGenericParameters();
+
+            var genericParametersNames = genericParameters
+                .Select(genericType => genericType.Name)
+                .ToArray();
+
+            var definedGenericParameters = defineGenericParameters(genericParametersNames);
+
+            for (var i = 0; i < genericParameters.Count; i++)
+            {
+                var genericParameter = genericParameters[i];
+                var definedGenericParameter = definedGenericParameters[i];
+                var genericParameterAttributes = genericParameter.GenericParameterAttributes
+                                                 & ~GenericParameterAttributes.Covariant
```

Comment:
Why exclude those?

Reply:
Because variant type parameters can be declared in interface or delegates only.

Reply:
Sure